marklogic-data-hub-central/src/main/java/com/marklogic/hub/central/auth/SessionMonitorInterceptor.java

```diff
@@ -0,0 +1,46 @@
+package com.marklogic.hub.central.auth;
+
+import org.jetbrains.annotations.NotNull;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.messaging.simp.SimpMessagingTemplate;
+import org.springframework.stereotype.Component;
+import org.springframework.web.servlet.HandlerInterceptor;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import java.util.HashMap;
+import java.util.Map;
+
+public class SessionMonitorInterceptor implements HandlerInterceptor {
+    static final Map<String, Object> stompHeaders = new HashMap<>() {{
+        put("content-type","application/json");
+    }};
+
+    private SimpMessagingTemplate template;
+
+    public SessionMonitorInterceptor(SimpMessagingTemplate template) {
+        this.template = template;
+    }
+
+    @Override
+    public boolean preHandle(
+            HttpServletRequest request, @NotNull HttpServletResponse response, @NotNull Object handler) throws Exception {
+        HttpSession session = request.getSession();
+        if (!(template == null || session == null)) {
+            int sessionTimeout = session.getMaxInactiveInterval();
```

Comment:
Why do we need a WebSocket to send the session timeout value continuously?
Wouldn't `getMaxInactiveInterval` always return 300 (5 minutes) in our case?

Reply:
I'm puzzled by that too - this is a constant value. I thought the intent was for the middle tier to send a message when the session expired, and/or when there are N seconds remaining before session expiration. 

An HttpSessionListener can be used for sending a message when the session expires. 

If there's a requirement for sending a message N seconds before expiration - I'm not sure how to do that. I could imagine some object we store in the HttpSession that waits until N seconds before expiration to send a message. That object would be overridden/reset every time a request is received.

Reply:
As @rjrudin points out, the HtttpSessionListener doesn't allow us to catch the situation where we are N seconds before closing the connection. The only opportunity to catch that is with each request. So with each request, I'm using the interceptor to reset the session count down on the front-end side. 

I'll make an update to only send out the update if the session last accessed time was more than a second ago to cut down on some of the chatter.